AzureMonitorAgent/agent.py

```diff
@@ -290,22 +290,28 @@ def install():
     config_updated = False
     try:
         if os.path.isfile(config_file):
-            data = ""
+            data = []
+            new_data = ""
+            export_dict = {}
+            vars_set = set()
             with open(config_file, "r") as f:
-                data = f.read()
-                for var in default_configs.keys():
-                    pattern = "(#*)export " + str(var) + "=.*$"
-                    r_obj = re.compile(pattern, flags=re.MULTILINE)
-                    value = "export " + var + "=" + default_configs[var]
-                    if var in data:
-                        new_data = r_obj.sub(value, data)
-                        data = new_data
-                    else:
-                        data += value+"\n"
+                data = f.readlines()
+                for line in data:
+                    for var in default_configs.keys():
+                        export_dict[var] = "export " + var + "=" + default_configs[var] + "\n"
+                        if var in line:
+                            line = export_dict[var]
+                            vars_set.add(var)
+                            break
+                    new_data += line
+            
+            for var in default_configs.keys():
+                if var not in vars_set:
+                    new_data += export_dict[var]
```

Comment:
this will be empty for new tokens not already in file in some edge cases due to break statement above

Reply:
I meant the export_dict[var]

Reply:
export_dict[var] will have an entry for every single var that is in default_configs. Any new tokens that are to be added will already be part of default_configs right? I am not sure if i understand the edge case where this will be empty. Could you give an example?